tree/dataframe/src/RLoopManager.cxx

```diff
@@ -133,11 +133,30 @@ void InsertBranchName(std::set<std::string> &bNamesReg, ColumnNames_t &bNames, c
 void ExploreBranch(TTree &t, std::set<std::string> &bNamesReg, ColumnNames_t &bNames, TBranch *b,
                           std::string prefix, std::string &friendName, bool allowDuplicates)
 {
+   // We want to avoid situations of overlap between the prefix and the
```

Comment:
Does the algorithm handle properly the case where (for better or worse), the valid branch name is something like `toplevel.fInner.fInner.fPt` (The top level class has a data member named `fInner` which is of a type that has a data member also named `fInner`)?  

Related but different:  The case where the user did not put the trailing dot and thus the toplevel name is `toplevel`, the branch name is `fInner.fInner.fPt` and the full branch name is stil `toplevel.fInner.fInner.fPt`.

Reply:
Just to clarify, this part of the code is never run based on user input. This is part of the routine that just populates a list of strings of all the valid branch names in the TTree, i.e. based on on-disk information alone. Thus, I believe the second point is not relevant here.

Regarding the first point, no, the algorithm now does not handle that properly. Unfortunately, I don't think there is a way to make the distinction. Crucially, without the current logic in the algorithm, the changes in TTree have no effect at all, i.e. we still have very poor performance in retrieving the full list of branch names. I need to create a reproducer first, then see what can be done on the TTree side to expose enough information to distinguish the two cases.

Reply:
After investigation, I can actually confirm that the logic is still working for the case of the nested data member with the same name. The reason is, it was faulty before already. I have created a test that is added to the latest commit. Before the logic is applied, the `fullName` variable is actually of the form `"toplevel.fInner.fInner.fInner.fPt"`, note the 3 repetitions of `fInner` instead of 2. So the logic removes the extra repetition. The real question is why TTree happily digests what looks to me like a wrong leaf name, but that is out of scope for this PR I believe.